src/entities/communitySbt.ts

```diff
@@ -174,6 +205,166 @@ class SBT {
         return totalSupply;
     }
 
+    public async getBadgeStatus(args: GetBadgesStatusArgs) : Promise<Array<BadgeWithStatus>| void> {
+        if (this.signer) {
+            const userAddress = await this.signer.getAddress();
+            const network = await this.provider?.getNetwork();
+            const networkName = network ? network.name : "";
+
+            const getURL = this.getEtherscanURL(networkName, args.apiKey, userAddress);
+            const resp = await axios.get(getURL);
+
+            const multiRedeemId = this.getMethodId(true, networkName);
+            const redeemId = this.getMethodId(false, networkName);
+
+            if (!resp.data) {
+                throw new Error("Etherscan api failed")
+            }
+            const transactions = resp.data.result;
+
+            // get last 50 transactions, match is redeem and set SUCC/FAILED status
+            let txBadges = new Map<number, TxBadgeStatus>();
+            for (const transaction of transactions) {
+                if (transaction.to.toLowerCase() === this.contract?.address.toLowerCase()) {
+                    const status = transaction.txreceipt_status === 1 ? 
+                        TxBadgeStatus.SUCCESSFUL
+                        : TxBadgeStatus.FAILED;
+                    if (transaction.methodId === redeemId) {
+                        const badgeType = this.decodeBadgeType(transaction.input);
+                        txBadges.set(badgeType, status);
+                    } else if (transaction.methodId === multiRedeemId) {
+                        const badgeTypes = this.decodeMultipleBadgeTypes(transaction.input);
+                        for (const badgeType of badgeTypes) {
+                            txBadges.set(badgeType, status);
+                        }
+                    }
+                    
+                }
+            }
+
+            // if badges of interest are not part of those 50 transactions, set them as pending
+            for (const badgeType of args.potenetialClaimingBadges) {
+                if (!txBadges.get(badgeType)) {
+                    txBadges.set(badgeType, TxBadgeStatus.PENDING);
+                }
+            }
+
+            // badges claiming status in subgraph - includes all bades earned by user in given season
+            const subgraphClaimedBadges = await this.claimedBadgesInSubgraph(args.subgraphUrl, userAddress, args.season);
+
+            // final claiming status verdict
+            const badgeStatus = subgraphClaimedBadges.map((badge) => {
```

Comment:
```suggestion
            const badgeStatuses = subgraphClaimedBadges.map((badge) => {
```

Reply:
changed in 4dfe51b4c9fc205a1a650ee996725346a3c7f545